pkg/liqo-controller-manager/resourceoffer-controller/resourceoffer_controller_methods.go

```diff
@@ -117,7 +118,7 @@ func (r *ResourceOfferReconciler) createVirtualKubeletDeployment(
 	// create the base resources
 	vkServiceAccount := forge.VirtualKubeletServiceAccount(name, namespace)
 	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, vkServiceAccount, func() error {
-		return controllerutil.SetOwnerReference(resourceOffer, vkServiceAccount, r.Scheme)
+		return nil
```

Comment:
Will this be deleted when removing the namespace?

Reply:
No, the Tenant Namespace where the peering-related resources are created (and also the VirtualKubelet) will not be deleted. The problem setting up this owner reference is that the ServiceAccount will be deleted before the VirtualKubelet termination. Removing it we will have a ServiceAccount with no related permission that will be used by the next VK in the next peering with that cluster.

In your opinion, is it better to leave this ServiceAccount or to delete it at the same time we delete the ClusterRoleBinding? (that is handled with a finalizer)

Reply:
The current approach seems fine to me, a serviceaccount without permissions does no harm.